esx_service/vmdk_ops.py

```diff
@@ -307,13 +307,14 @@ def listVMDK(path):
 
 # Return VM managed object, reconnect if needed. Throws if fails twice.
 def findVmByUuid(vm_uuid):
+    vm = None
     try:
         vm = si.content.searchIndex.FindByUuid(None, vm_uuid, True, False)
-        if vm:
-            return vm
-    except Exception as e:
-        logging.exception("Failed to find VM uuid=%s (traceback below), " \
-                          "retrying...", vm_uuid)
+    except Exception as ex:
```

Comment:
there is no guarantee, but we know that on connection timeout retry fixes it, and on other issues we want to rethrow if we fail in retry